Automattic/woocommerce-payments
# Fix error while changing payment method of subscription product

Fixes #3924 

#### Changes proposed in this Pull Request

Fix error in updating subscription when saved cards are more than `posts per page` setting

* Pass a large `token limit` argument to the `get_tokens` function, to mitigate the error. 
* This is a short-term solution to avoid the error for an edge case, where number of saved cards is greater than `posts per page`
* The longer term solution will be to fix this open issue in WC Core. - 
https://github.com/woocommerce/woocommerce/issues/25025





#### Testing instructions



1.   In the `wp_options` table set `posts_per_page` row to `1`
2.  Make sure your account has more than 1 saved card with WC Payments
3.  Purchase a new subscription with a new saved card
4.  Visit the Edit Subscription admin page for this new subscription (**WooCommerce > Subscriptions > click on subscription ID**)
5.  Update the status of the subscription from Active to On-hold
6.  Press Update button on the Edit Subscription page.
7. You should see no errors, and only a message mentioning the subscription status has been updated
8. If you try to edit the user payment details within the sub in WP Admin, you should be able to see the other payment methods in it.  



#### Before  

a. Error on changing status of subscription
![image](https://user-images.githubusercontent.com/4162931/186721939-b4394882-6941-485c-a903-6b0d3e40b90e.png)


b. No payment methods appearing within the Subscription Billing settings
![image](https://user-images.githubusercontent.com/4162931/186722082-06370345-aa13-4aaa-a9aa-345384d978ed.png)

#### After

a. On changing status of subscription ( no more error ) 
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/4162931/186722601-c83bd923-a273-4e63-abec-6341e2527b7d.png">

b. Payment methods appear in dropdown in subscription billing settings 
<img width="1217" alt="image" src="https://user-images.githubusercontent.com/4162931/186722868-9aef44e8-eb3f-413b-99f2-a461e89cc173.png">


-------------------

- [x] Tested on mobile (or does not apply) - Not applicable

**Post merge**
